Packages/com.unity.inputsystem/InputSystem/Editor/Debugger/SampleFrequencyCalculator.cs

```diff
@@ -0,0 +1,48 @@
+using UnityEngine.InputSystem.LowLevel;
+
+namespace UnityEngine.InputSystem.Editor
+{
+    /// <summary>
+    /// A utility for computing sample frequency based on input events.
+    /// </summary>
+    internal struct SampleFrequencyCalculator
+    {
+        private double m_LastUpdateTime;
+        private int m_SampleCount;
+
+        public SampleFrequencyCalculator(float targetFrequency, double realtimeSinceStartup)
```

Comment:
Target frequency is currently only a "convient storage" here. Hower I would suggest keeping it inside this type since an evolved implementation will likely use it to compute error compared to target and create a CDF or PDF of it.